### DISCUSSION TITLE
Regarding Leaderboard system on the discord server

### DISCUSSION THREAD

**Discussion by Vyvy-vi** [EddieHubCommunity/support#544] [General] [answered]
The current way MEE6(the discord bot) ranks users on the EddieJaoude discord server, might not prove to be an inclusive measure(now the levelling system is another thing, which I think is alright), but the user ranking and the leaderboard might cause some pseudo sense of persuing some "X" rank in new users and might lead to a non-inclusive environment. (The rankings do not represent any form of expertise or superiority, thus portrayal of ranks make it sort of unnecessary) 

### Additional Context:
![Screenshot 2020-12-11 at 8 28 39 PM](https://user-images.githubusercontent.com/62864373/101918560-7f116100-3bef-11eb-889d-179de25e6dad.png)
https://mee6.xyz/leaderboard/699608417039286293

**Comment by Akshu-on-github** [comment]
What if we reset the bot every month? Or in a similar time frame?

**Reply by Vyvy-vi** [reply]
hmm interesting prospect...
This way the ranking would rather represent: Most active in the community this week/month, which would be a more sensible metric 🤔

**Reply by SKULLXL** [reply]
> hmm interesting prospect...
> This way the ranking would rather represent: Most active in the community this week/month, which would be a more sensible metric 🤔

We can rather use weekly points then. I forgot If mee6 has this feature Amari bot does I think mee6 also does but it's a premium feature.

**Comment by SKULLXL** [comment]
I disagree cause having a leaderboard may make people be more active and kind of encourage them to get into more convos and help people with open source and coding in general. Plus having a leaderboard means we can check into who is the most talky person in the server and is getting into convos. Plus the point that the newcomers may feel inclusive, for this I think we can do what @AkshayaKulasekaran suggested. Thats just my opinion 🤷🏻‍♂️, I may be wrong tho.

**Comment by BOLT04** [comment]
@Vyvy-vi I understand your concern, and the topic of user rankings and leaderboards is tough. It would take some time to implement a system that incentivizes open source contributions in order to rank up (we have this similar issue: https://github.com/EddieJaoudeCommunity/EddieBot/issues/32).

However, I also agree with @SKULLXL on the fact that we can look at the leaderboard to check users who are more involved in the server, in regards to messages. But you are right, this isn't really a good metric because it doesn't mean a particular person is helping someone in the community...

IMO, I'm in favor of figuring out how to do @AkshayaKulasekaran proposed solution, then create an issue for it. Unless we decide to build something better for user ranks that is more aligned with our community 😃

**Comment by eddiejaoude** [comment]
It is difficult to find a balance and I think it is great we are all trying 👍 

Let's take small steps in the direction, what could we do first and fairly easily? Maybe reset MEE6 bot manually monthly for now? Anything else?

**Reply by Akshu-on-github** [reply]
I think that sounds good. Maybe look into adding a monthly point system to Eddiebot?

**Reply by Vyvy-vi** [reply]
I think for a start, resetting MEE6 monthly would be a great step in this direction 👍

**Reply by eddiejaoude** [reply]
Monthly reset is a good idea, I wonder if we can do that with a GitHub Action to an API using a scheduler?

**Reply by eddiejaoude** [reply]
I manually reset everyone's XP 😱 

Here it is before the reset...

<img width="1426" alt="Screenshot 2020-12-28 at 08 36 30" src="https://user-images.githubusercontent.com/624760/103201456-dd288e80-48e7-11eb-994d-314b428b3cb8.png">

**Reply by Vyvy-vi** [reply]
I don't know about automating the XP leaderboard. Perhaps we could automate it using selenium? 
afaik, they don't have an API from where you could access xp stats

**Reply by eddiejaoude** [reply]
I am happy to manually reset it for now to keep things simple. It is only 12 times per year, so not a big issue :)

**Comment by Akshu-on-github** [comment]
Yup yup... Oh, on the topic of badges, why don't we have a badge for each reset? 1st reset, 2nd reset, etc?

It would lessen the impact of the feeling of "losing" points while being a fun way to keep track of the growth of the community!

**Reply by Vyvy-vi** [reply]
I didn't get, what kind of badges are you suggesting? 😅 
Could you give some more detail? Is is it like, we keep a log of past leaderboard too?

**Reply by Akshu-on-github** [reply]
Hmm, like... **This user has been a part of the community since the nth reset!** or, **This user was present for the nth reset!**

**Reply by Vyvy-vi** [reply]
that would be great 🤔

**Reply by eddiejaoude** [reply]
I think it effectively would be how many months since they joined starting from this month (if we reset every month from now)?

**Reply by Vyvy-vi** [reply]
I think we won't need to store that
(Discord stores server join date, so that shouldn't be an issue 🤔 )

**Comment by Akshu-on-github** [comment]
Yes, although it could also show the months they weren't active if we limit the badges to those who have non-zero XP that month

**Reply by eddiejaoude** [reply]
good point 👍

**Reply by Akshu-on-github** [reply]
Whoops, meant to reply to the previous comment 😅

**Comment by stemount** [comment, marked as answer]
Is there really an answer to this question do we think? I am wondering if we should lock/archive this? It sounds a bit 50:50 on "reset every month" vs "lets gamify all the things!"

![](http://4.bp.blogspot.com/-LeW5R4hxd_k/T1kkaYdT85I/AAAAAAAAEP8/U-v1dB7uFwI/s1600/memes_not_sure_if_all_the_things_or_just_the_similar_things_memebase_89-s500x375-220214.jpg)

**Reply by Akshu-on-github** [reply]
I believe so, it seems like the conversation has shifted to badges which we can discuss in the other issue

**Reply by Vyvy-vi** [reply]
🤣  I suppose we should
